monai/apps/deepgrow/transforms.py

```diff
@@ -0,0 +1,462 @@
+        probability: key that represents click/interaction probability.
+        batched: whether input is batched or not.
+    """
+
+    def __init__(
+        self,
+        guidance: str = "guidance",
+        discrepancy: str = "discrepancy",
+        probability: str = "probability",
+        batched: bool = True,
+    ):
+        self.guidance = guidance
+        self.discrepancy = discrepancy
+        self.probability = probability
+        self.batched = batched
+
+    def randomize(self, data=None):
+        pass
+
+    def find_guidance(self, discrepancy):
+        distance = distance_transform_cdt(discrepancy).flatten()
+        probability = np.exp(distance) - 1.0
+        idx = np.where(discrepancy.flatten() > 0)[0]
+
+        if np.sum(discrepancy > 0) > 0:
+            seed = self.R.choice(idx, size=1, p=probability[idx] / np.sum(probability[idx]))
+            dst = distance[seed]
+
+            g = np.asarray(np.unravel_index(seed, discrepancy.shape)).transpose().tolist()[0]
+            g[0] = dst[0]
+            return g
+        return None
+
+    def add_guidance(self, discrepancy, probability):
+        will_interact = self.R.choice([True, False], p=[probability, 1.0 - probability])
+        if not will_interact:
+            return None, None
+
+        pos_discr = discrepancy[0]
+        neg_discr = discrepancy[1]
+
+        can_be_positive = np.sum(pos_discr) > 0
+        can_be_negative = np.sum(neg_discr) > 0
+        correct_pos = np.sum(pos_discr) >= np.sum(neg_discr)
+
+        if correct_pos and can_be_positive:
+            return self.find_guidance(pos_discr), None
+
+        if not correct_pos and can_be_negative:
+            return None, self.find_guidance(neg_discr)
+        return None, None
+
+    def _apply(self, guidance, discrepancy, probability):
+        guidance = guidance.tolist() if isinstance(guidance, np.ndarray) else guidance
+        if not self.batched:
+            pos, neg = self.add_guidance(discrepancy, probability)
+            if pos:
+                guidance[0].append(pos)
+                guidance[1].append([-1] * len(pos))
+            if neg:
+                guidance[0].append([-1] * len(neg))
+                guidance[1].append(neg)
+        else:
+            for g, d, p in zip(guidance, discrepancy, probability):
+                pos, neg = self.add_guidance(d, p)
+                if pos:
+                    g[0].append(pos)
+                    g[1].append([-1] * len(pos))
+                if neg:
+                    g[0].append([-1] * len(neg))
+                    g[1].append(neg)
+        return np.asarray(guidance)
+
+    def __call__(self, data):
+        d = dict(data)
+        guidance = d[self.guidance]
+        discrepancy = d[self.discrepancy]
+        probability = d[self.probability]
+
+        d[self.guidance] = self._apply(guidance, discrepancy, probability)
+        return d
+
+
+class SpatialCropForegroundd(MapTransform):
```

Comment:
could this be merged into https://github.com/Project-MONAI/MONAI/blob/master/monai/transforms/croppad/dictionary.py#L367 with some non-breaking changes?
otherwise please clearly highlight the difference in the docstring

Reply:
added highlight in docstring